smolyay/smolyak.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Jul  5 08:00:15 2022
+
+@author: mzf0069
+"""
+
+import numpy
+import scipy.special
+
+
+class IndexGrid:
+    """Indexes of grid points for constructing a surrogate function.
+
+    Parameters
+    ----------
+    dimension : int
+        Number of independent variables.
+    exactness : int
+        Level of exactness of the approximation.
+    index_per_level : list
+        Number of indexes at each level.
+    """
+
+    def __init__(self, dimension, exactness, index_per_level):
+        self.dimension = dimension
+        self.exactness = exactness
+        self.index_per_level = index_per_level
```

Comment:
You will want to turn each of these into a property as well, which has both a getter and a setter. The setters should make `self._needs_update = True`.